From 8155f78d76e6b059a7390edc06ad7cf432a737e6 Mon Sep 17 00:00:00 2001
From: ajayk <ajaykemparaj@gmail.com>
Date: Mon, 2 Jun 2025 04:06:24 -0700
Subject: [PATCH] Fix version header generation by replacing non-portable hex
 escape

Replace \x23 hexadecimal escape with \# in Makefile.am to fix
version.h generation on systems where sed doesn't support \x23.
This resolves build failures where JQ_VERSION is undefined.
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index b6ac671..78f3271 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -99,7 +99,7 @@ endif
 # Remake the version.h header file if, and only if, the git ID has changed
 .PHONY: .FORCE
 .FORCE:
-generate_ver = ver="`{ $(srcdir)/scripts/version || echo '$(VERSION)' ; } | sed 's/.*/\x23define JQ_VERSION \"&\"/'`"
+generate_ver = ver="`{ $(srcdir)/scripts/version || echo '$(VERSION)' ; } | sed 's/.*/\#define JQ_VERSION \"&\"/'`"
 .remake-version-h: .FORCE
 	@ $(generate_ver); test "x`cat src/version.h 2>/dev/null`" = "x$$ver" || touch .remake-version-h
 src/version.h: .remake-version-h
-- 
2.39.5 (Apple Git-154)

